core/src/main/java/hudson/util/io/ZipArchiver.java

```diff
@@ -94,6 +94,7 @@ public void visit(final File f, final String _relativePath) throws IOException {
             ZipEntry fileZipEntry = new ZipEntry(this.prefix + relativePath);
             if (mode!=-1)   fileZipEntry.setUnixMode(mode);
             fileZipEntry.setTime(f.lastModified());
+            fileZipEntry.setSize(f.length());
```

Comment:
With the fix to the test case but without this change to `src/main`, the test fails with "`org.apache.tools.zip.Zip64RequiredException`: `huge64bitFileTest.txt`'s size exceeds the limit of 4GByte" on OpenJDK 11.0.10. With the fix to the test case and with this change to `src/main`, the test passes on OpenJDK 11.0.10.